docs/pages/technical-specs/expo-updates-0.md

```diff
@@ -0,0 +1,187 @@
+
+The following describes how a client-library will interact with a conforming updates server:
+1. The client-library will make a [request](#request) for the most recent manifest. 
+2. If it is a new manifest, the client-library will proceed to make requests to download and store the assets specified in the manifest.
+3. Finally, the client-library will update its local state according to any metadata provided by the response [headers](#headers).
+
+We anticipate the primary user of this spec will be organizations who need to manage their own update server to satisfy internal requirements.
+
+## Manifest Request
+A conformant client-library MUST make a GET request with the following headers:
+```
+accept: application/json
+expo-platform: string
+expo-runtime-version: string
+```
+along with any headers stipulated by a previous responses' [server defined headers](#headers):
+
+* `expo-platform` MUST be platform type the client is running on: 
+    * iOS MUST be `expo-platform: ios`
+    * Android MUST be `expo-platform: android`
+* `expo-runtime-version` MUST be the runtime version the client is running on.
+
+A conformant client-library SHOULD also accept `application/expo+json` so that servers can easily tell when to send back an Expo manifest:
+```
+accept: application/expo+json, application/json
+```
+
+## Manifest Response
+
+A conformant server MUST return a body containing a [manifest](#body) along with the [headers](#headers) specified below.
+
+The choice of manifest and headers are dependent on the values of the request headers. A conformant server MUST:
+
+* Return a manifest that describes an update capable of running on the platform and runtime version specified in the `expo-platform` and `expo-runtime-version` request headers. 
+* Return a manifest that describes the most recent, sorted by creation time, update satisfying any constraints imposed by the headers.
+
+### Headers
+
+```
+expo-protocol-version: 0
+expo-sfv-version: 0
+expo-manifest-filters: expo-sfv
+expo-server-defined-headers: expo-sfv
+cache-control: *
+content-type: application/json; charset=utf-8
+```
+
+* `expo-protocol-version` describes the version of the protocol defined in this spec and MUST be `0`.
+* `expo-sfv-version`  MUST be `0`.
+* `expo-manifest-filters` is an [Expo SFV 0](expo-sfv-0.md) dictionary. It is used to filter updates stored by the client-library by the `updateMetadata` attributes found in the [manifest](#body).
+  * For example: `expo-manifest-filters: branchname="main"` instructs the client-library to load the most recent update it has stored whose `updateMetadata` contains:
+
+  ```
+  updateMetadata: {
+    branchname: 'main',
+    ...
+  }
+  ```
+  * If the `branchname` manifest filter is included, it MUST equal the `branchName` in the `manifest.updateMetadata`.
```

Comment:
the lowercasing is intentional because HTTP structured field values, which we use to define the manifest filters, allow only lowercase strings. that said we could benefit from explicitly noting the difference in cases here.

we wanted to keep branchName camelcased to be consistent with our JSON field naming

(also this is an implementation detail but I think `manifest.updateMetadata.bRaNcHnAmE` will match the manifest filters since expo-updates does a case-insensitive comparison. however other parts of the code might expect `branchName`)